### ISSUE TITLE
Old versioned jars are not deleted

### ISSUE THREAD

**Issue by boopac** [edvin/fxlauncher#133]
Hi,
if i update a library (controlsfx-8.40.13 -> controlsfx-8.40.14) the old jar files are not deleted.

With maven i build the new app.xml which contain the tag for controlsfx-8.40.14.jar.
I copy the new controlsfx-8.40.14.jar to the URL and delete controlsfx-8.40.13.jar.

But after using fxlauncher I have both files on the client:
controlsfx-8.40.13.jar (not deleted)
controlsfx-8.40.14.jar (new downloaded)

Is there a possibility to delete the fiels that are not in app.xml with fxlauncher?

King regards,
Boris

**Comment by mordechaim** [comment]
No you can't but it will not be dynamically loaded onto the JVM if it
is not present in the manifest.

if you really want, your artifacts should not have a version string so
newer will overwrite older. In case it's already released, rename the
new version to match the older.

**Comment by edvin** [comment]
Thanks @mordechaim - I agree. We could have added a feature to delete old artifacts, but that's a bit dangerous, and since we can remove the need for it by simply removing version numbers that's a safer approach IMO.

**Comment by boopac** [comment]
@mordechaim: that's right, but then you have a lot of garbage files, if the artifacts have new versions.

@edvin: if you remove the version number, you can't see on client site if there is the right versions of the artifacts.

Maybe in a later version of fxlauncher there is a parameter for "--remove-old-artifacts".

**Comment by boopac** [comment]
That's would be pretty cool :+1:

**Comment by mordechaim** [comment]
You will have to add more entries in the manifest, namely 'deletes'.
Otherwise how would we know what's an old artifact?

Do you think it's still in the scope of this framework? One could
easily do the deletes theirselves in the actual app; it doesn't need
to be done anywhere before the launch process. Just my opinion.

**Comment by edvin** [comment]
We'd have to have a designated artifacts-folder so it would be safe for FXLauncher to delete any files not mentioned in the manifest. This could be done after the new version has been downloaded successfully.
